content/en/BestPractices/defend-against-ssrf.md

```diff
@@ -0,0 +1,213 @@
+---
+title: Protect against Server-Side Request Forgery
+linkTitle: Protect against SSRF
+weight: 620
+description: >
+  Protect your app against SSRF attacks.
+---
+
+{{% pageinfo %}}
+See examples of Server-Side Request Forgery (SSRF) exploitation, and learn how to minimize your risks.
+{{% /pageinfo %}}
+
+SSRF is ranked 10<sup>th</sup> in the [OWASP Top 10 Web Application Security Risks (2021)](https://owasp.org/www-project-top-ten/) based on its impact and likelihood.
+
+## Description
+
+[Server-Side Request Forgery (SSRF)](https://owasp.org/Top10/A10_2021-Server-Side_Request_Forgery_%28SSRF%29/) is a server-side security vulnerability that allows an attacker to make arbitrary requests from the application server.
+
+SSRF has a wide range of impacts. For example, it may allow a threat actor to perform a simple external interaction to steal sensitive metadata information, which may lead to account takeover. Read a real-life [case study](https://medium.com/swlh/capital-one-data-breach-a-cloud-security-case-study-7a06ec900460) describing this type of attack.
+
+A malicious actor may achieve the following through a successful SSRF exploitation:
+
+- Scan a local or remote port
+
+- Read a local file
+
+- Interact with internal apps, services, or network
+
+- Read metadata from cloud platforms such as Amazon Web Services (AWS), Microsoft Azure, Google Cloud, or DigitalOcean
+
+- Perform attacks:
+
+  - Remote Code Execution (RCE) by chaining services on the internal network
+
+  - Reflected Cross-Site Scripting (XSS)
+
+  - Cross-Site Request Forgery (CSRF)
+
+## Attack Scenarios
+
+Let's see some examples of SSRF attacks.
+
+### Accessing a Localhost
+
+In the following code example, the application accepts user-supplied input through the `url` parameter and passes it to the `urllib.request.urlopen(url).read()` function. The application processes URLs without filtration, which may lead to an SSRF attack.
+
+An attacker may pass a URL such as `https://www.target.com/test/?url=http://127.0.0.1` to the vulnerable code and access the localhost `127.0.0.1`.
```

Comment:
Let's avoid "Naming names". Per RFC 2606, we can use generic names like example.com for documentation. So I suggest:

```suggestion
An attacker may pass a URL such as `https://www.example.com/test/?url=http://127.0.0.1` to the vulnerable code and access the localhost `127.0.0.1`.
```

Reply:
I'm going to _assume_ this change is OK, as we'd like to plan to "publish". @harsh-bothra, let us know if you see a problem.